data-prepper-plugins/cloudwatch-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/configuration/NamespaceConfig.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.List;
+
+/**
+ *  Configuration class to get the Namespace info
+ */
+public class NamespaceConfig {
+
+    @JsonProperty("name")
+    @NotNull
+    private String name;
+    @JsonProperty("start_time")
+    @NotNull
+    private String startTime;
+    @JsonProperty("end_time")
+    @NotNull
+    private String endTime;
+    @JsonProperty("metricDataQueries")
+    @NotNull
+    private List<MetricDataQueriesConfig> metricDataQueriesConfig;
```

Comment:
Similar to my other comment about YAML simplification. We can have:

```
metricDataQueries:
  - name:                   
     id: "q1"
     period: 
  - name: 
     id: "q2"
     period: 
     stat: "Average"
```

To resolve:

1. Change this line to: `private List<MetricsConfig> metricsConfigs;`
2. Change the getter and all corresponding usages.
3. Delete `MetricDataQuerieisConfig` entirely.

Reply:
this has been fixed.